### PR TITLE
Add close button to the post resolve reminder

### PR DESCRIPTION
**Pull request by jedrek0429**
https://github.com/Programowanie-Discord/ProgramowanieBot/pull/5#discussion_r1186710559

### CODE REVIEW

File: ProgramowanieBot/Handlers/MessageHandler.cs [hunk 1]
```diff
@@ -98,10 +99,7 @@ Task HandleReminderAsync()
                     return Task.CompletedTask;
 
                 var content = message.Content;
-                if (Config.PostResolveReminderKeywords.Any(k => content.Contains(k, StringComparison.InvariantCultureIgnoreCase)))
-                    return SendReminderMessageAsync();
-
-                return Task.CompletedTask;
+                return Config.PostResolveReminderKeywords.Any(k => content.Contains(k, StringComparison.InvariantCultureIgnoreCase)) ? SendReminderMessageAsync() : Task.CompletedTask;
```
Comment (jedrek0429) [inline ProgramowanieBot/Handlers/MessageHandler.cs]: I decided to include this small change. Forgot to mention it in the commit and PR description. Apologies.

File: ProgramowanieBot/Handlers/MessageHandler.cs [hunk 2]
```diff
@@ -122,7 +120,21 @@ await using (var context = Provider.GetRequiredService<DataContext>())
...
+                            Components = new ActionRowProperties[]
+                            {
+                                new(new ButtonProperties[]
+                                {
+                                    new ActionButtonProperties($"close:{thread.OwnerId}", Config.PostCloseButtonLabel, ButtonStyle.Danger),
+                                }),
+                            },
+                            MessageReference = new(message.Id),
+                        };
+                        await thread.SendMessageAsync(messageProperties);
```
Comment (KubaZ2) [inline ProgramowanieBot/Handlers/MessageHandler.cs]:
```suggestion
                        await thread.SendMessageAsync(new()
                        {
                            Content = Config.PostResolveReminderMessage,
                            Components = new ActionRowProperties[]
                            {
                                new(new ButtonProperties[]
                                {
                                    new ActionButtonProperties($"close:{thread.OwnerId}", Config.PostCloseButtonLabel, ButtonStyle.Danger),
                                }),
                            },
                            MessageReference = new(message.Id),
                        });
```

File: ProgramowanieBot/Handlers/MessageHandler.cs (hunk 1 above)
Comment (KubaZ2) [inline ProgramowanieBot/Handlers/MessageHandler.cs]: I considered this formatting before, but I think using `if` is clearer and looks better

Comment (jedrek0429) [reply ProgramowanieBot/Handlers/MessageHandler.cs]: Okay. So be it.

Comment (jedrek0429) [reply ProgramowanieBot/Handlers/MessageHandler.cs]: https://github.com/Programowanie-Discord/ProgramowanieBot/pull/5#discussion_r1186711588

Comment (jedrek0429) [inline ProgramowanieBot/Handlers/MessageHandler.cs]:
```suggestion
               if (Config.PostResolveReminderKeywords.Any(k => content.Contains(k, StringComparison.InvariantCultureIgnoreCase)))
                    return SendReminderMessageAsync();

                return Task.CompletedTask;
```

File: ProgramowanieBot/Handlers/MessageHandler.cs [hunk 3]
```diff
@@ -98,7 +99,7 @@ Task HandleReminderAsync()
                     return Task.CompletedTask;
 
                 var content = message.Content;
-                if (Config.PostResolveReminderKeywords.Any(k => content.Contains(k, StringComparison.InvariantCultureIgnoreCase)))
+               if (Config.PostResolveReminderKeywords.Any(k => content.Contains(k, StringComparison.InvariantCultureIgnoreCase)))
```
Comment (jedrek0429) [inline ProgramowanieBot/Handlers/MessageHandler.cs]:
```suggestion
                if (Config.PostResolveReminderKeywords.Any(k => content.Contains(k, StringComparison.InvariantCultureIgnoreCase)))
```

File: ProgramowanieBot/Handlers/MessageHandler.cs (hunk 1 above)
Comment (jedrek0429) [reply ProgramowanieBot/Handlers/MessageHandler.cs]: https://github.com/Programowanie-Discord/ProgramowanieBot/pull/5/commits/8cc2143743f4deb13c4bb18d4f53eaedc5564cde

**Review by KubaZ2 — APPROVED** [review]

**Pull request merged**